btf/types.go

```diff
@@ -36,6 +37,9 @@ type Type interface {
 	// Make a copy of the type, without copying Type members.
 	copy() Type
 
+	// Compare the Type against another. Ignores Name and any descendant Types.
+	equal(Type) bool
```

Comment:
Why is ignoring names the right thing to do?

Reply:
For my purposes, struct/type/func layouts were the lowest-common denominator comparing kernel types. e.g. names are not considered to determine if a func satisfies a C prototype or Go interface.